### PR TITLE
[DOP-3396]: Guides search index jobs are not enqueuing

### PR DESCRIPTION
**Pull request by branberry**
## Ticket
[DOP-3396](https://jira.mongodb.org/browse/DOP-3396)

## Test Plan
1. Go to the [following build](https://workerpoolstaging-qgeyp.mongodbstitch.com/pages/job.html?collName=dotcom_queue&jobId=63cad941f604c938c792a809) and verify that the build is successful.
2.  Verify that the [search manifest job is enqueued](https://workerpoolstaging-qgeyp.mongodbstitch.com/pages/job.html?collName=dotcom_queue&jobId=63cadabb0ca3b75c00f89464)
3.  Verify that the [search manifest job is executed successfully](https://workerpoolstaging-qgeyp.mongodbstitch.com/pages/job.html?collName=dotcom_queue&jobId=63cadabb0ca3b75c00f89464)

### CODE REVIEW

File: src/job/jobHandler.ts [hunk 1]
```diff
@@ -387,35 +387,48 @@ export abstract class JobHandler {
...
+        this.currJob._id,
+        `${this._config.get<string>('stage').padEnd(15)}Pushing to ${this.name}`
+      );
 
-    if ((this.currJob?.deployCommands?.length ?? 0) > 0) {
-      const resp = await this._commandExecutor.execute(this.currJob.deployCommands);
-      if (resp?.error?.includes('ERROR')) {
+      if ((this.currJob?.deployCommands?.length ?? 0) > 0) {
+        const resp = await this._commandExecutor.execute(this.currJob.deployCommands);
+        if (resp?.error?.includes?.('ERROR')) {
```
Comment (branberry) [inline src/job/jobHandler.ts:399]: the fix is here

**Comment by branberry** [comment]
Converting this back to draft to address test failures

File: src/job/jobHandler.ts [hunk 2]
```diff
@@ -387,35 +387,46 @@ export abstract class JobHandler {
...
-        this.currJob._id,
-        `${this._config.get<string>('stage').padEnd(15)}Pushing to ${
-          this.name
-        } failed as there is no commands to execute`
-      );
-      throw new PublishError(`Failed pushing to ${this.name}, No commands to execute`);
+    } catch (error) {
+      if (error instanceof Error) {
+        const errorMessage = error.message;
+        throw new Error(`Caught the following error when performing generic deploy ${errorMessage}`);
```
Comment (seungpark) [inline src/job/jobHandler.ts:427]: think we can propagate the original error here. can log, but shouldn't throw a different error from what has been caught

**Review by seungpark — CHANGES_REQUESTED** [review]
minor change but overall LGTM. great debugging and stack tracing!

File: src/job/jobHandler.ts (hunk 2 above)
Comment (branberry) [reply src/job/jobHandler.ts:427]: got it, will update!

**Review by seungpark — APPROVED** [review]

**Pull request merged**